### DISCUSSION TITLE
Control log_dict's on_epoch log name

### DISCUSSION THREAD

**Discussion by johanneskvamme** [Lightning-AI/pytorch-lightning#7209] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hey!

I tried to change to using the new way of logging through `MetricCollection` and `self.log_dict` instead of logging every metric through `self.log` on `training step` and `test_epoch_end`. However, each metric is then logged as `[metric_name]_epoch_[epoch_number]` which creates a new graph for every epoch instead of allowing me to use epoch on the x-axis of my graphs (on Comet, if that is relevant). Is there a way to control this behaviour of` log_dic`t, or do I just have to keep logging "manually" to control log name?

**Comment by SkafteNicki** [comment, marked as answer]
Hi @FluidSense,

Could you try just updating the `MetricCollection` during the `_step` method and then log in `_epoch_end` method. Something like:
```python
def training_step(self, batch, batch_idx):
    logits = self(x)
    self.train_metrics.update(logits, y)

def train_epoch_end(self, outputs):
    self.log_dict(self.train_metrics.compute())
```

**Reply by johanneskvamme** [reply]
@SkafteNicki Thanks for the suggestion! I'll try that first thing in the morning and come back to you here afterwards

**Reply by johanneskvamme** [reply]
This works great for us at least! It takes a bit more time to get the first indications of how things are going, but at least the metrics are gathered correctly.
